Git doc: Create new doc for git summary

Aerosane opened the pull request:
While git is the most important part of any project in open source, we don't have a doc summarising our workflow.

Therefore,
Add a doc to explain the workflow.

Aerosane committed:
Update git_summary.md to remove redundant text

jayatheerthkulkarni commented:
Hey, there are some changes you can make here.

1. You have to use 

git commit -s

So that your name is signed off below the commit (I think the doc has git commit -m in your patch, you can fix that too)

2. If you read the first two docs in beginners, you will find the first line to very crisply summarise what will the doc contain, we call it _before you delve in_

3. You can take a tutorial, a small test case and show the demonstration of each of these commands in the docs, I think that would be fine.

4. The commit message itself, the main heading says removed redundant things, well you didn't remove anything here, it's a new doc right? So fix that, and the description of the commit message is fair enough.

Apart from that this is pretty good to merge, and I really like the conversational style here.

jayatheerthkulkarni commented:
A good commit message would be,

---

Git doc: Create new doc for git summary

While git is the most important part of any project in open source, we don't have a doc summarising our workflow.

Therefore,
Add a doc to explain the workflow.

Signed off <Name>
---
You get signed off only when you use

_git commit -s_
and not 
_git commit -m_

This follows the rule set in the documentation of the beginners. There is a very specific rule mentioned there.

Thank you
- Jayatheerth

jayatheerthkulkarni committed:
Git doc: Create new doc for git summary

While git is the most important part of any project in open source, we don't have a doc summarising our workflow.

Therefore,
Add a doc to explain the workflow.
